src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3590,6 +3590,11 @@ private void AddSynthesizedMembers(MembersAndInitializersBuilder builder, Declar
                 case TypeKind.Submission:
                     AddSynthesizedTypeMembersIfNecessary(builder, declaredMembersAndInitializers, diagnostics);
                     AddSynthesizedConstructorsIfNecessary(builder, declaredMembersAndInitializers, diagnostics);
+
+                    if (TypeKind == TypeKind.Class)
```

Comment:
nit: consider tightening this check to only top-level non-generic static classes